pkg/component/kubernetes/scheduler/scheduler.go

```diff
@@ -168,319 +207,293 @@ func (k *kubeScheduler) Deploy(ctx context.Context) error {
 		command              = k.computeCommand(port)
 	)
 
-	if err := k.client.Create(ctx, configMap); client.IgnoreAlreadyExists(err) != nil {
-		return err
-	}
-
-	if _, err := controllerutils.GetAndCreateOrMergePatch(ctx, k.client, service, func() error {
-		service.Labels = getLabels()
-
-		utilruntime.Must(gardenerutils.InjectNetworkPolicyAnnotationsForScrapeTargets(service, networkingv1.NetworkPolicyPort{
-			Port:     ptr.To(intstr.FromInt32(port)),
-			Protocol: ptr.To(corev1.ProtocolTCP),
-		}))
-
-		service.Spec.Selector = getLabels()
-		service.Spec.Type = corev1.ServiceTypeClusterIP
-		desiredPorts := []corev1.ServicePort{{
-			Name:     portNameMetrics,
-			Protocol: corev1.ProtocolTCP,
-			Port:     port,
-		}}
-		service.Spec.Ports = kubernetesutils.ReconcileServicePorts(service.Spec.Ports, desiredPorts, corev1.ServiceTypeClusterIP)
-
-		return nil
-	}); err != nil {
-		return err
-	}
-
-	if err := shootAccessSecret.Reconcile(ctx, k.client); err != nil {
-		return err
-	}
-
-	if _, err := controllerutils.GetAndCreateOrMergePatch(ctx, k.client, deployment, func() error {
-		deployment.Labels = utils.MergeStringMaps(getLabels(), map[string]string{
-			v1beta1constants.GardenRole:                                         v1beta1constants.GardenRoleControlPlane,
-			resourcesv1alpha1.HighAvailabilityConfigType:                        resourcesv1alpha1.HighAvailabilityConfigTypeController,
-			v1beta1constants.LabelExtensionProviderMutatedByControlplaneWebhook: "true",
-		})
-		deployment.Spec.Replicas = &k.replicas
-		deployment.Spec.RevisionHistoryLimit = ptr.To[int32](1)
-		deployment.Spec.Selector = &metav1.LabelSelector{MatchLabels: getLabels()}
-		deployment.Spec.Template = corev1.PodTemplateSpec{
-			ObjectMeta: metav1.ObjectMeta{
-				Labels: utils.MergeStringMaps(getLabels(), map[string]string{
-					v1beta1constants.GardenRole:                 v1beta1constants.GardenRoleControlPlane,
-					v1beta1constants.LabelPodMaintenanceRestart: "true",
-					v1beta1constants.LabelNetworkPolicyToDNS:    v1beta1constants.LabelNetworkPolicyAllowed,
-					gardenerutils.NetworkPolicyLabel(v1beta1constants.DeploymentNameKubeAPIServer, kubeapiserverconstants.Port): v1beta1constants.LabelNetworkPolicyAllowed,
-				}),
+	// Service Resource
+	service.Labels = getLabels()
+	utilruntime.Must(gardenerutils.InjectNetworkPolicyAnnotationsForScrapeTargets(service, networkingv1.NetworkPolicyPort{
+		Port:     ptr.To(intstr.FromInt32(port)),
+		Protocol: ptr.To(corev1.ProtocolTCP),
+	}))
+	service.Spec.Selector = getLabels()
+	service.Spec.Type = corev1.ServiceTypeClusterIP
+	desiredPorts := []corev1.ServicePort{{
+		Name:     portNameMetrics,
+		Protocol: corev1.ProtocolTCP,
+		Port:     port,
+	}}
+	service.Spec.Ports = kubernetesutils.ReconcileServicePorts(service.Spec.Ports, desiredPorts, corev1.ServiceTypeClusterIP)
```

Comment:
You don't need `emptyService()` function now, you can directly construct the resource.